perun-core/src/main/java/cz/metacentrum/perun/core/implApi/modules/attributes/UserVirtualAttributeCollectedFromUserExtSource.java

```diff
@@ -157,39 +160,66 @@ public Attribute getAttributeValue(PerunSessionImpl sess, User user, AttributeDe
 	}
 
 	/**
-	 * Get list of message patterns used for resolving attribute value change.
-	 * @return List of Patterns
+	 * Functional interface controlling AuditEvents
```

Comment:
This JavaDoc should be more verbose. You should state here, what it should do and what it is used for. Also, you should add javadoc to the `check` method. You should state there, that the method is used to determine whether some auditevent should be handled. And that is represented by returning user id. If it should not be handled, null is returned.